### ISSUE TITLE
UDP Socket Time Out

### ISSUE THREAD

**Issue by ebnerjoh** [grebleem/WeatherLinkliveUDP#14]
Hi, 

I installed this package for my Davis Vantage Pro 2 Wireless with the Weatherlink Live.

I am getting an Error Message with "UDP Socket Time Out, but strangly everything is working.
```

Sep 20 08:32:57 weewx-test weewx[1217] ERROR user.weatherlinkliveudp: UDP Socket Time Out
Sep 20 08:32:57 weewx-test weewx[1217] INFO user.weatherlinkliveudp: UDP check at: 2020-09-20 09:32:57 CEST (1600587177)
```

What does this error message mean?

Br,
Johannes

**Comment by grebleem** [comment]
Hi,

It looks like UDP is not running on the wll, can you tell me a little more about your setup?

**Comment by ebnerjoh** [comment]
Hi,

I just unpacked the WLL, linked it to my wireless lan and added the wll
driver to weewx and configured the IP.

Not sure if and how I could enable UDP on the WLL?

Br,
Johannes

Am Mi., 23. Sept. 2020 um 20:59 Uhr schrieb grebleem <
notifications@github.com>:

**Comment by grebleem** [comment]
The driver should enable the UDP automatically.
Do you have the Davis Weatherlink on your phone? If you open the app it should give 'Last Updated: Live' on the top of the screen. that is also an indication UDP broadcast is on.

**Comment by ebnerjoh** [comment]
Hi, 

Yes, the Mobile App is showing "Live"

**Comment by grebleem** [comment]
That is good, where is weewx running? Perhaps a firewall is blocking UDP input.

**Comment by ebnerjoh** [comment]
I have it running on an RPi with Weewx 4.1.1

There is no FW between Weewx and the WLL.

**Comment by ebnerjoh** [comment]
Strange, I am not seeing this errror anymore...

**Comment by c0by85** [comment]
I had the same error 1 - 3 times. Disappears again by itself. Currently the error no longer occurs. Nothing changed.

```
root@pi2 ~ # cat /var/log/syslog | grep weewx | grep -i error


root@pi2 ~ # cat /var/log/syslog | grep weewx | grep -i udp
Sep 24 00:45:18 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 01:45:18 CEST (1600904718)
Sep 24 01:40:48 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 02:40:48 CEST (1600908048)
Sep 24 02:36:18 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 03:36:18 CEST (1600911378)
Sep 24 03:31:48 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 04:31:48 CEST (1600914708)
Sep 24 04:27:18 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 05:27:18 CEST (1600918038)
Sep 24 05:22:49 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 06:22:49 CEST (1600921369)
Sep 24 06:18:18 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 07:18:18 CEST (1600924698)
Sep 24 07:13:48 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 08:13:48 CEST (1600928028)
Sep 24 08:09:18 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 09:09:18 CEST (1600931358)
Sep 24 09:04:48 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 10:04:48 CEST (1600934688)
```

**Comment by grebleem** [comment]
Thanks for the info. UDP can be a bit tricky, it have to be 'switched on' in the wll. That is the UDP check you see in the log. 
I think that ebnerjoh had no UDP data for some reason. When it rains the driver has to to some calculations and runs into an error because of missing UDP data. I'll have to make the driver a bit more solid on that.
For me the driver is running for more than a few month now with no errors (and a lot of rain ;-) ).
At the moment I am a bit busy with another project, but this will definitely go on the To Fix list.

**Comment by ebnerjoh** [comment]
Thanks, I can also confirm that I see this error only from time to time

**Comment by c0by85** [comment]
strange...

```
root@pi2 ~ # cat /var/log/syslog | grep weewx | grep -i error
Sep 24 14:19:21 pi2 weewx[377] ERROR user.weatherlinkliveudp: UDP Socket Time Out
Sep 24 14:19:25 pi2 weewx[377] WARNING urllib3.connectionpool: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPConnectionPool(host='192.168.1.7', port=80): Read timed out. (read timeout=3)")': /v1/real_time?duration=3600
Sep 24 14:19:30 pi2 weewx[377] WARNING urllib3.connectionpool: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPConnectionPool(host='192.168.1.7', port=80): Read timed out. (read timeout=3)")': /v1/real_time?duration=3600
Sep 24 14:19:37 pi2 weewx[377] WARNING urllib3.connectionpool: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPConnectionPool(host='192.168.1.7', port=80): Read timed out. (read timeout=3)")': /v1/real_time?duration=3600
Sep 24 14:19:40 pi2 weewx[377] ERROR user.weatherlinkliveudp: RequestExeption: HTTPConnectionPool(host='192.168.1.7', port=80): Max retries exceeded with url: /v1/real_time?duration=3600 (Caused by ReadTimeoutError("HTTPConnectionPool(host='192.168.1.7', port=80): Read timed out. (read timeout=3)"))
Sep 24 14:19:40 pi2 weewx[377] ERROR user.weatherlinkliveudp: Unable to connect to Weather Link Live
Sep 24 14:19:45 pi2 weewx[377] ERROR user.weatherlinkliveudp: UDP Socket Time Out
Sep 24 14:19:48 pi2 weewx[377] WARNING urllib3.connectionpool: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPConnectionPool(host='192.168.1.7', port=80): Read timed out. (read timeout=3)")': /v1/real_time?duration=3600
Sep 24 14:19:53 pi2 weewx[377] WARNING urllib3.connectionpool: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPConnectionPool(host='192.168.1.7', port=80): Read timed out. (read timeout=3)")': /v1/real_time?duration=3600
Sep 24 14:20:00 pi2 weewx[377] WARNING urllib3.connectionpool: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x6f023650>, 'Connection to 192.168.1.7 timed out. (connect timeout=3)')': /v1/real_time?duration=3600
```

```
root@pi2 ~ # cat /var/log/syslog | grep weewx | grep -i udp
Sep 24 00:45:18 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 01:45:18 CEST (1600904718)
Sep 24 01:40:48 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 02:40:48 CEST (1600908048)
Sep 24 02:36:18 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 03:36:18 CEST (1600911378)
Sep 24 03:31:48 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 04:31:48 CEST (1600914708)
Sep 24 04:27:18 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 05:27:18 CEST (1600918038)
Sep 24 05:22:49 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 06:22:49 CEST (1600921369)
Sep 24 06:18:18 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 07:18:18 CEST (1600924698)
Sep 24 07:13:48 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 08:13:48 CEST (1600928028)
Sep 24 08:09:18 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 09:09:18 CEST (1600931358)
Sep 24 09:04:48 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 10:04:48 CEST (1600934688)
Sep 24 10:00:18 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 11:00:18 CEST (1600938018)
Sep 24 10:55:49 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 11:55:49 CEST (1600941349)
Sep 24 11:51:20 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 12:51:20 CEST (1600944680)
Sep 24 12:46:49 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 13:46:49 CEST (1600948009)
Sep 24 13:42:19 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 14:42:19 CEST (1600951339)
Sep 24 14:19:21 pi2 weewx[377] ERROR user.weatherlinkliveudp: UDP Socket Time Out
Sep 24 14:19:40 pi2 weewx[377] ERROR user.weatherlinkliveudp: RequestExeption: HTTPConnectionPool(host='192.168.1.7', port=80): Max retries exceeded with url: /v1/real_time?duration=3600 (Caused by ReadTimeoutError("HTTPConnectionPool(host='192.168.1.7', port=80): Read timed out. (read timeout=3)"))
Sep 24 14:19:40 pi2 weewx[377] ERROR user.weatherlinkliveudp: Unable to connect to Weather Link Live
Sep 24 14:19:45 pi2 weewx[377] ERROR user.weatherlinkliveudp: UDP Socket Time Out
Sep 24 14:20:00 pi2 weewx[377] INFO user.weatherlinkliveudp: UDP check at: 2020-09-24 15:20:00 CEST (1600953600)
```

i had to log in at weatherlink.com and see if there was a new firmware.

Checked...

```
WeatherLink LIVE Firmware Version
Jul 30, 2020
```

**Comment by grebleem** [comment]
Fixed in v0.2.9.
